### PR TITLE
RPN-750 add check-asset-type to sell and bid methods

### PR DESCRIPTION
**Pull request by vitalbadjo**

### CODE REVIEW

File: src/order/bid.ts [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
 	Part,
 } from "@rarible/protocol-api-client"
 import { randomWord, toBigNumber } from "@rarible/types"
 import { toBn } from "../common/to-bn"
+import {AssetTypeRequest, AssetTypeResponse} from "./check-asset-type";
 
 export type BidRequest = {
 	maker: Address
-	makeAssetType: EthAssetType | Erc20AssetType,
+	makeAssetType: AssetType,
```
Comment (evgenynacu) [inline src/order/bid.ts:14]: That's not correct, I think, should be something like Erc721Type | Erc1155Type | { contract: Address, tokenId: BigNumber }

File: src/order/check-asset-type.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+} from "@rarible/protocol-api-client";
+
+export type NftAssetType = {
+    contract: Address;
+    tokenId: BigNumber;
+}
+
+export type AssetTypeRequest = Erc721AssetType | Erc1155AssetType | NftAssetType
+
+export type AssetTypeResponse = Exclude<AssetTypeRequest, NftAssetType>
```
Comment (evgenynacu) [inline src/order/check-asset-type.ts:16]: Seems like Erc721AssetType | Erc1155AssetType

File: src/order/check-asset-type.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+export async function checkAssetType(nftItemApi: NftItemControllerApi, collectionApi: NftCollectionControllerApi, asset: AssetTypeRequest): Promise<AssetTypeResponse> {
```
Comment (evgenynacu) [inline src/order/check-asset-type.ts:19]: I think we need test for this function. You can for example, use rinkeby and check ERC721/ERC1155 collections there

**Review by evgenynacu — CHANGES_REQUESTED** [review]

File: src/order/check-asset-type.ts (hunk 2 above)
Comment (vitalbadjo) [reply src/order/check-asset-type.ts]: updated

File: src/order/check-asset-type.ts (hunk 3 above)
Comment (vitalbadjo) [reply src/order/check-asset-type.ts:19]: test done

**Pull request merged**